### PR TITLE
Have SQUID_CHECK_LIB_WORKS do state SAVE/RESTORE

### PR DESCRIPTION
**Pull request by yadij**
Removing a lot of duplicated code and further
simplifying library detection.

### CODE REVIEW

**Review by kinkie — APPROVED** [review]
Good idea!

**Review by rousskov — CHANGES_REQUESTED** [review]

File: acinclude/squid-util.m4 [hunk 1]
```diff
@@ -224,7 +224,9 @@ dnl   and ensure that it is empty or unset (on failures).
 AC_DEFUN([SQUID_CHECK_LIB_WORKS],[
 AH_TEMPLATE(m4_toupper(m4_translit([HAVE_LIB$1], [-+.], [___])),[Define as 1 to enable '$1' library support.])
 AS_IF([m4_translit([test "x$with_$1" != "xno"], [-+.], [___])],[
+  SQUID_STATE_SAVE(check_lib_works_state)
   $2
+  SQUID_STATE_ROLLBACK(check_lib_works_state)
   AS_IF([! test -z m4_toupper(m4_translit(["$LIB$1_LIBS"], [-+.], [___]))],[
```
[the remark this answers is not in the document]
Comment (yadij) [reply acinclude/squid-util.m4:230]: I spent today checking that one to verify.
 * It does appear that the comment a few lines above about `LT_INIT pulling it in somehow is correct. I was not able to detect any linker differences by removing this `LIBS=` update.
 * The `LIBDL_PATH` is already placed into `CPPFLAGS`, so  no effect from removing the global change to `LDFLAGS`

File: acinclude/squid-util.m4 [hunk 2]
```diff
@@ -224,7 +224,9 @@ dnl   and ensure that it is empty or unset (on failures).
 AC_DEFUN([SQUID_CHECK_LIB_WORKS],[
 AH_TEMPLATE(m4_toupper(m4_translit([HAVE_LIB$1], [-+.], [___])),[Define as 1 to enable '$1' library support.])
 AS_IF([m4_translit([test "x$with_$1" != "xno"], [-+.], [___])],[
+  SQUID_STATE_SAVE(check_lib_works_state)
```
[the remark this answers is not in the document]
Comment (yadij) [reply acinclude/squid-util.m4:230]: Done in [41e29e9](https://github.com/squid-cache/squid/pull/1774/commits/41e29e9d1f64688ca209ace9c29e29aa89a3499b)

**Review by rousskov — CHANGES_REQUESTED** [review]
Almost done!

**Review by rousskov — APPROVED** [review]

**Pull request closed without merging**